engines/native/system/user_darwin.go

```diff
@@ -38,9 +40,24 @@ func CurrentUser() (*User, error) {
 		panic(fmt.Sprintf("Could not convert %s to integer: %s", osUser.Gid, err))
 	}
 
+	// Find group ids
+	sgids, err := osUser.GroupIds()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to list groups for current user")
+	}
+	gids := make([]uint32, len(sgids))
+	for i, sgid := range sgids {
+		igid, err := strconv.Atoi(sgid)
+		gids[i] = uint32(igid)
```

Comment:
Shouldn't we assign after error checking? I know in practice it doesn't matter in this case, but "check error first, assign the value later" is a good coding practice.